order-service/order-service.ts

```diff
@@ -0,0 +1,238 @@
+import { Tx, Pair, Side, Position, OrderConfig } from '../types/common.js';
+import {
+  OrderType,
+  BECH32_PREFIX,
+  LocalWallet,
+  SubaccountClient,
+  Network,
+  CompositeClient,
+  IndexerClient,
+} from '@dydxprotocol/v4-client-js';
+
+interface OrderService {
+  placeLimitOrder: (
+    pair: Pair,
+    side: Side,
+    price: number,
+    size: number,
+    config: OrderConfig,
+  ) => Promise<Tx>;
+  placeMarketOrder: (
+    pair: Pair,
+    side: Side,
+    price: number,
+    size: number,
+    config: OrderConfig,
+  ) => Promise<Tx>;
+  placeLimitTakeProfitOrder: (
+    pair: Pair,
+    side: Side,
+    price: number,
+    size: number,
+    config: OrderConfig,
+  ) => Promise<Tx>;
+  placeMarketTakeProfitOrder: (
+    pair: Pair,
+    side: Side,
+    price: number,
+    size: number,
+    config: OrderConfig,
+  ) => Promise<Tx>;
+  placeLimitStopLossOrder: (
+    pair: Pair,
+    side: Side,
+    price: number,
+    size: number,
+    config: OrderConfig,
+  ) => Promise<Tx>;
+  placeMarketStopLossOrder: (
+    pair: Pair,
+    side: Side,
+    price: number,
+    size: number,
+    config: OrderConfig,
+  ) => Promise<Tx>;
+  listPositions: () => Promise<Position[]>;
+}
+
+// order factory function
+const createOrderService = (
+  mnemonic: string,
+  network: Network,
+): OrderService => ({
+  placeLimitOrder: (pair, side, price, size, config) => {
+    return new Promise<Tx>(async (resolve) => {
```

Comment:
Mark the function as `async`, and you don't need to return a promise explicitly. It's much more readable.

Reply:
It's not what I mean.
Instead of
```ts
function foo() {
  return new Promise((res) => {
    // some stuff
    res(someValue);
  });
}
```
You can simply do
```ts
async function foo() {
  // some stuff
  return someValue;
}
```
There are rare cases that you have to use the Promise constructor syntax, but this is not one of them.